src/vim9class.c

```diff
@@ -1904,5 +1904,75 @@ object_free_nonref(int copyID)
     return did_free;
 }
 
+/*
+ * Return TRUE when the class "cl", its base class or one of the implemented interfaces
+ * matches the class "other_cl".
+ */
+    int
+class_instance_of(class_T *cl, class_T *other_cl)
+{
+    if (cl == other_cl)
+	return TRUE;
+
+    // Recursively check the base classes.
+    for (; cl != NULL; cl = cl->class_extends)
+    {
+	if (cl == other_cl)
+	    return TRUE;
+	// Check the implemented interfaces.
+	for (int i = cl->class_interface_count - 1; i >= 0; --i)
+	    if (cl->class_interfaces_cl[i] == other_cl)
+		return TRUE;
+    }
+
+    return FALSE;
+}
+
+/*
+ * "instanceof(object, classinfo)" function
+ */
+    void
+f_instanceof(typval_T *argvars, typval_T *rettv)
+{
+    typval_T	*object_tv = &argvars[0];
+    typval_T	*classinfo_tv = &argvars[1];
+    listitem_T	*li;
+
+    rettv->vval.v_number = FALSE;
+
+    if (object_tv->v_type != VAR_OBJECT || object_tv->vval.v_object == NULL)
```

Comment:
Instead of this check, code similar to the following is needed here:

```
    if (check_for_object_arg(argvars, 0) == FAIL
                || check_for_class_or_list_arg(argvars, 1) == FAIL)
        return;
```

The functions check_for_object_arg() and check_for_class_or_list_arg() should be added to typval.c.
With this check in place, the last else statement below is not needed.